### DISCUSSION TITLE
how to convert Jbrowse 1 plugin to Jbrowse 2

### DISCUSSION THREAD

**Discussion by mthang** [GMOD/jbrowse-components#4187] [Q&A]
I have used this plugin named ComboTrackSelector (https://github.com/Arabidopsis-Information-Portal/ComboTrackSelector) for my Jbrowse1 and I would like to find out how to get this plugin working on Jbrowse2. Or is it possible to convert this plugin for Jbrowse2 ?

**Comment by cmdcolin** [comment]
Hi there,
JBrowse 1 plugins are not compatible with JBrowse 2 in general, JBrowse 2 is a big rewrite of JBrowse 1 and doesn't maintain total backwards compatibility. 

However, in this case, JBrowse 2 actually took inspiration from ComboTrackSelector so by default, the track selector for JBrowse 2 has both the hierarchical and the faceted track selector available by default.

Screenshots:

Default track selector

![image](https://github.com/GMOD/jbrowse-components/assets/6511937/a641f68d-4cb1-4708-aa8b-1579486d9fc7)

Menu showing "Open faceted track selector" on desktop

![image](https://github.com/GMOD/jbrowse-components/assets/6511937/953fc95d-d5ed-4889-bace-a05314606d70)

resulting faceted track selector

![image](https://github.com/GMOD/jbrowse-components/assets/6511937/8a2212fd-a068-4263-85f7-7d6aece68bd9)

**Comment by mthang** [comment]
@cmdcolin Thank you for the quick response ! Great to know that Jbrowse2 has included the facet track selector component. By looking at the Faceted track selector (last snapshot above). I would like to know how to customise the metadata table to suit my need. I have a separate csv file that containing all the metadata and need to define a list of header inside the trackList.json file to show the metadata table on ComboTrackSelector. I wonder what's the configuration look like in the config.json file. do you have an example file that you can share ?

**Reply by cmdcolin** [reply]
There is no trackMetadata.csv support in JBrowse 2, so instead, you add a "metadata" section to each track config.

example

```json
{
  "type": "AlignmentsTrack",
  "trackId": "mytrack",
  "name": "My track",
  "metadata": {
    "prep": "flash frozen",
    "sequencing": "Illumina PE",
    "source": "150bp PE reads from a <i>Volvox mythicus</i>. See <a href=https://google.com>Google</a>"
  },
  "assemblyNames": ["volvox"],
  "adapter": {
    "type": "CramAdapter",
    "cramLocation": {
      "uri": "volvox-sorted-altname.cram"
    },
    "craiLocation": {
      "uri": "volvox-sorted-altname.cram.crai"
    },
    "sequenceAdapter": {
      "type": "TwoBitAdapter",
      "twoBitLocation": {
        "uri": "volvox.2bit"
      }
    }
  }
}

```

this would then make three columns for "prep", "sequencing", and "source" in the faceted track selector automatically. There is a user interface option that actually "hides" columns automatiacally that aren't shared across many tracks (the "Show sparse metadata columns") so you can toggle that on and off

![image](https://github.com/GMOD/jbrowse-components/assets/6511937/b64de169-04c1-49dd-9730-debba48d086e)

so basically, to make the trackMetadata.csv work with JBrowse 2, you would probably have to write a script that reads in the csv, and then populates the config.json with the metadata obejct for each track

**Reply by mthang** [reply]
This is what I thought how the tracks would be generated "so basically, to make the trackMetadata.csv work with JBrowse 2, you would probably have to write a script that reads in the csv, and then populates the config.json with the metadata obejct for each track" . Thanks!

**Reply by cmdcolin** [reply]
yep!
